tensorflow_datasets/image/stanford_dogs.py

```diff
@@ -0,0 +1,194 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow as tf
+import tensorflow_datasets.public_api as tfds
+import xml.etree.ElementTree as ET
+from collections import defaultdict
+import os
+import re
+
+_DESCRIPTION = '''\
+The Stanford Dogs dataset contains images of 120 breeds of dogs from around
+the world. This dataset has been built using images and annotation from
+ImageNet for the task of fine-grained image categorization. There are
+20,580 images, out of which 12,000 are used for training and 8580 for
+testing. Class labels and bounding box annotations are provided
+for all the 12,000 images.
+'''
+
+_URL = ("http://vision.stanford.edu/aditya86/ImageNetDogs/main.html")
+
+_CITATION = """\
+@inproceedings{KhoslaYaoJayadevaprakashFeiFei_FGVC2011,
+author = "Aditya Khosla and Nityananda Jayadevaprakash and Bangpeng Yao and Li Fei-Fei",
+title = "Novel Dataset for Fine-Grained Image Categorization",
+booktitle = "First Workshop on Fine-Grained Visual Categorization, IEEE Conference on Computer Vision and Pattern Recognition",
+year = "2011",
+month = "June",
+address = "Colorado Springs, CO",
+}
+@inproceedings{imagenet_cvpr09,
+        AUTHOR = {Deng, J. and Dong, W. and Socher, R. and Li, L.-J. and Li, K. and Fei-Fei, L.},
+        TITLE = {{ImageNet: A Large-Scale Hierarchical Image Database}},
+        BOOKTITLE = {CVPR09},
+        YEAR = {2009},
+        BIBSOURCE = "http://www.image-net.org/papers/imagenet_cvpr09.bib"}
+"""
+
+_IMAGES_URL = "http://vision.stanford.edu/aditya86/ImageNetDogs/images.tar"
+_SPLIT_URL = "http://vision.stanford.edu/aditya86/ImageNetDogs/lists.tar"
+_ANNOTATIONS_URL = "http://vision.stanford.edu/aditya86/ImageNetDogs/annotation.tar"
+_NAME_RE = re.compile(r"([\w-]*/)*([\w]*.jpg)$")
+
+
+class StanfordDogs(tfds.core.GeneratorBasedBuilder):
+    '''Stanford Dogs dataset'''
+
+    VERSION = tfds.core.Version("0.1.0")
+
+    def _info(self):
+
+        return tfds.core.DatasetInfo(
+            builder=self,
+            description=_DESCRIPTION,
+            features=tfds.features.FeaturesDict({
+                # Images are of varying size
+                "image": tfds.features.Image(),
+                "image/filename": tfds.features.Text(),
+                "label": tfds.features.ClassLabel(num_classes=120),
+                # Multiple bounding box per image
+                "objects": tfds.features.SequenceDict({
+                    "bbox": tfds.features.BBoxFeature(),
+                }),
+            }),
+            supervised_keys=("image", "label"),
+            urls=_URL,
+            citation=_CITATION
+        )
+
+    def _split_generators(self, dl_manager):
+
+        download_path = dl_manager.download([_SPLIT_URL, _ANNOTATIONS_URL, _IMAGES_URL])

```

Comment:
Yes, they are needed, because I need the dataset annotation and split files to be extracted out after downloading so that I can parse the contents in them. 
However, the folder containing the images can remain as an archive and can be iterated over using dl_manager.iter_archive() for fetching the examples